nixos/modules/services/misc/forgejo.nix

```diff
@@ -616,19 +622,21 @@ in
         # System Call Filtering
         SystemCallArchitectures = "native";
         SystemCallFilter = [ "~@cpu-emulation @debug @keyring @mount @obsolete @privileged @setuid" "setrlimit" ];
+      } // lib.optionalAttrs (cfg.settings.server.SSH_PORT < 1024) {
+        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
+        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE";
+        PrivateUsers = false;
```

Comment:
Hmm there are actually a lot of config options that can trigger a bind to a port below 1024.

For SSH, `server.SSH_LISTEN_PORT` defaults to whatever `server.SSH_PORT` is (which is UI-only) and only applies when `server.START_SSH_SERVER` is true and `server.DISABLE_SSH` is false (I think).

And then there is also `server.ENABLE_ACME`, `server.PORT_TO_REDIRECT`, `server.REDIRECT_OTHER_PORT` and `server.HTTP_PORT` and whatnot with their own dependency chain.

I changed my mind and believe we should simply set `CAP_NET_BIND_SERVICE` unconditionally.
The security implications are tiny, and saves us from re-implementing upstream's config logic for all those special cases (which, to add insult to insure, may very well change with an update any time).

```diff
diff --git a/nixos/modules/services/misc/forgejo.nix b/nixos/modules/services/misc/forgejo.nix
index 34668cb2e16c..1f85a3865a8e 100644
--- a/nixos/modules/services/misc/forgejo.nix
+++ b/nixos/modules/services/misc/forgejo.nix
@@ -596,7 +596,8 @@ in
         ReadWritePaths = [ cfg.customDir cfg.dump.backupDir cfg.repositoryRoot cfg.stateDir cfg.lfs.contentDir ];
         UMask = "0027";
         # Capabilities
-        CapabilityBoundingSet = "";
+        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" ];
+        CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" ];
         # Security
         NoNewPrivileges = true;
         # Sandboxing
@@ -604,7 +605,7 @@ in
         ProtectHome = true;
         PrivateTmp = true;
         PrivateDevices = true;
-        PrivateUsers = true;
+        PrivateUsers = false; # true is incompatible with our set of AmbientCapabilities
         ProtectHostname = true;
         ProtectClock = true;
         ProtectKernelTunables = true;
@@ -622,10 +623,6 @@ in
         # System Call Filtering
         SystemCallArchitectures = "native";
         SystemCallFilter = [ "~@cpu-emulation @debug @keyring @mount @obsolete @privileged @setuid" "setrlimit" ];
-      } // lib.optionalAttrs (cfg.settings.server.SSH_PORT < 1024) {
-        AmbientCapabilities = "CAP_NET_BIND_SERVICE";
-        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE";
-        PrivateUsers = false;
       };
 
       environment = {
```

Reply:
Updated this!